extensions/wikidata/src/org/openrefine/wikidata/qa/WikidataConstraintFetcher.java

```diff
@@ -340,4 +336,84 @@ public boolean usableOnItems(PropertyIdValue pid) {
         }
         return results;
     }
+
+    protected List<QuantityValue> getValues(List<SnakGroup> groups, String pid) {
+        List<QuantityValue> results = new ArrayList<>();
+        for (SnakGroup group : groups) {
+            if (group.getProperty().getId().equals(pid)) {
+                for (Snak snak : group.getSnaks())
+                    results.add((QuantityValueImpl) snak.getValue());
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Is this property expected to have a value whose difference
+     * with it's lower bound property should be in a range?
+     */
+    @Override
+    public boolean hasDiffWithinRange(PropertyIdValue pid) {
+        return getSingleConstraint(pid, DIFFERENCE_WITHIN_RANGE_CONSTRAINT_QID) != null;
+    }
+
+    /**
+     * Retrieves the lower value property for calculating the difference
+     * required in difference-within-range constraint
+     *
+     * @param pid:
+     *            the property to calculate difference with
+     * @return the pid of the lower bound property
+     */
+    @Override
+    public PropertyIdValue getLowerPropertyId(PropertyIdValue pid) {
```

Comment:
We have the same problem as the Conflicts With constraint here: as a parameter, we should pass the target property id (the one against which the difference is checked), and then find the corresponding constraint definition instead of only reading the first one.